pkg/v2/json/deserialize.go

```diff
@@ -133,6 +133,31 @@ Skip:
 	return string(d.data[start+1 : end-1]), nil
 }
 
+// For the given dot-separated path - return an array of simple names in stack order.
+// These names could be consumed by the prop.Navigator.Dot() method.
+// For example for the path = "name.givenName"
+// return ["givenName", "name"]
+func (d *deserializeState) getPropertyDotNamesByPath(path string) []string {
+	var names []string
+	var findComplexPath func(child prop.Property) bool
```

Comment:
minor, you can combine this line and below to avoid duplicating signatures:

```
findComplexPath := func(child prop.Property) bool {
...
```

Reply:
btw, are there unit tests in this library? this would be a good candidate

Reply:
@vocoded 
> minor, you can combine this line and below to avoid duplicating signatures:
> 
> ```
> findComplexPath := func(child prop.Property) bool {
> ...
> ```

It won't work here; as there's the recursive call to `findComplexPath` within itself

Reply:
@vocoded 
> btw, are there unit tests in this library? this would be a good candidate

Added the test for this case